testutil/testutil.go

```diff
@@ -21,15 +21,137 @@ import (
 	"reflect"
 	"sort"
 
-	"github.com/openconfig/gnmi/value"
-
+	"github.com/golang/protobuf/proto"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
 	gnmipb "github.com/openconfig/gnmi/proto/gnmi"
+	"github.com/openconfig/gnmi/value"
 )
 
-// pathLess provides a function which determines whether a gNMI Path messages
+// NotificationSetEqual compares the contents of a and b and returns true if
+// they are equal. Order of the slices is ignored.
+func NotificationSetEqual(a, b []*gnmipb.Notification) bool {
+	return cmp.Equal(a, b, cmpopts.SortSlices(NotificationLess))
+}
+
+// UpdateSetEqual compares the contents of a and b and returns true if they are
+// equal. Order of the slices is ignored.
+func UpdateSetEqual(a, b []*gnmipb.Update) bool {
+	return cmp.Equal(a, b, cmpopts.SortSlices(UpdateLess))
+}
+
+// updateSet is an alias for a slice of gNMI Update messages.
+type updateSet []*gnmipb.Update
+
+// Len, Less, and Swap implement the sort.Interface interface.
+func (u updateSet) Len() int           { return len(u) }
```

Comment:
Thanks -- I looked at this, and having the type and methods here, rather than defining the inline functions in multiple places seemed cleaner. Will leave as is at the moment.